stl/inc/utility

```diff
@@ -110,14 +110,30 @@ _NODISCARD constexpr _Ty(min)(initializer_list<_Ty>); // implemented in <algorit
 
 _EXPORT_STD template <class _Ty, size_t _Size, enable_if_t<_Is_swappable<_Ty>::value, int> /* = 0 */>
 _CONSTEXPR20 void swap(_Ty (&_Left)[_Size], _Ty (&_Right)[_Size]) noexcept(_Is_nothrow_swappable<_Ty>::value) {
-    if (&_Left != &_Right) {
-        _Ty* _First1 = _Left;
-        _Ty* _Last1  = _First1 + _Size;
-        _Ty* _First2 = _Right;
-        for (; _First1 != _Last1; ++_First1, ++_First2) {
-            swap(*_First1, *_First2); // intentional ADL
+    if (&_Left == &_Right) {
+        // Indirect precondiition:
+        // N4988 [utility.swap]/7: Effects: As if by swap_ranges.
+        // N4988 [alg.swap]/2: Precondition: The two ranges [first1, last1) and [first2, last2) do not overlap.
+
+        return; // Loosening this precondition and allow self-swap; see LWG-4165

```

Comment:
Hmm, if the precondition will gets relaxed, `swap` shouldn't return here unless it's proven that the swapping is no-op. Self-swap can have some side effects and we shouldn't skip them if any, see #4597.

Reply:
Hmm, OK to me. I think it's also reasonable to make such change when LWG-4165 gets resolved.

Reply:
We then need to update this comment

Reply:
Ah, you already did that